test/fuzz/fuzz_runner.h

```diff
@@ -11,6 +11,19 @@
 namespace Envoy {
 namespace Fuzz {
 
+// Each test may need a sub-environment of that provided by //test/test_common:environment_lib,
+// since each fuzz invocation runs in the same process, but might want a distinct tmp sandbox for
+// example.
+class PerTestEnvironment {
+public:
+  PerTestEnvironment();
+
+  std::string temporaryPath(const std::string& path) const { return test_tmpdir_ + "/" + path; }
+
+private:
+  std::string test_tmpdir_;
```

Comment:
nit: You could make `test_tmpdir_` const and initialize in the ctor initializer list:

```
PerTestEnvironment::PerTestEnvironment() : test_tmpdir_([](){
  const std::string fuzz_path = TestEnvironment::temporaryPath("fuzz_XXXXXX");
  char test_tmpdir[fuzz_path.size() + 1];
  StringUtil::strlcpy(test_tmpdir, fuzz_path.data(), fuzz_path.size() + 1);
  RELEASE_ASSERT(::mkdtemp(test_tmpdir) != nullptr, "");
  return std::string(test_tmpdir);
}()) {}
```

Reply:
I've had people point out to me in the past that this makes C++ look like JavaScript spaghetti, but sure, why not :)